src/app/features/asset-list/bitcoin/runes-asset-list/runes-asset-list.tsx

```diff
@@ -11,20 +15,49 @@ interface RuneTokenAssetDetails {
 
 interface RunesAssetListProps {
   runes: RuneTokenAssetDetails[];
+  rightElementVariant: RightElementVariant;
+  hasRunesSetter?: React.Dispatch<React.SetStateAction<boolean>>;
 }
-export function RunesAssetList({ runes }: RunesAssetListProps) {
-  return runes.map((rune, i) => (
-    <CryptoAssetItemLayout
-      availableBalance={createMoneyFromDecimal(
-        convertAmountToBaseUnit(rune.balance.availableBalance),
-        rune.info.symbol,
-        rune.info.decimals
-      )}
-      balanceSuffix={rune.info.symbol}
-      captionLeft="Runes"
-      icon={<RunesAvatarIcon />}
-      key={`${rune.info.symbol}${i}`}
-      titleLeft={rune.info.spacedRuneName ?? rune.info.runeName}
-    />
-  ));
+
+export function RunesAssetList({
+  runes,
+  rightElementVariant,
+  hasRunesSetter,
+}: RunesAssetListProps) {
+  useEffect(() => {
+    if (hasRunesSetter && runes.length) hasRunesSetter(true);
+  }, [runes.length, hasRunesSetter]);
+
+  if (!runes.length) return null;
+
+  const icon = <RunesAvatarIcon />;
+  return runes.map((rune, i) => {
+    const titleLeft = rune.info.spacedRuneName ?? rune.info.runeName;
+    const captionLeft = 'Runes';
+    const symbol = rune.info.symbol;
+    if (rightElementVariant === 'toggle') {
+      return (
+        <CryptoAssetItemToggleLayout
+          captionLeft={captionLeft}
+          titleLeft={titleLeft}
+          key={`${symbol}`}
```

Comment:
You can just use `key={symbol}` here instead but it could be good to also add the `index`